src/compiler/checker.ts

```diff
@@ -79,17 +79,24 @@ namespace ts {
             getDeclaredTypeOfSymbol,
             getPropertiesOfType,
             getPropertyOfType,
+            getIndexInfoOfType,
             getSignaturesOfType,
             getIndexTypeOfType,
             getBaseTypes,
+            getUnionType,
+            getIntersectionType,
+            getTypeFromTypeReference,
             getReturnTypeOfSignature,
             getNonNullableType,
             getSymbolsInScope,
+            createSymbol,
             getSymbolAtLocation,
             getShorthandAssignmentValueSymbol,
             getExportSpecifierLocalTargetSymbol,
             getTypeAtLocation: getTypeOfNode,
             getPropertySymbolOfDestructuringAssignment,
+            signatureToString,

```

Comment:
`signatureToString` is needed independently in the checker anyways, so I'd prefer to avoid copying the code if it can be avoided. 

`indexSignatureToString`: Only used in one place in the current PR, but we may find it useful in a future codefix for adding index signatures. Will inline for now, but may have to re-declare it in the future.